core/src/test/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectCreatorSpec.scala

```diff
@@ -0,0 +1,105 @@
+    0,
+    RawlsUserSubjectId("sub"),
+    None)
+
+
+  behavior of "validateBillingProjectCreationRequest"
+
+  it should "fail when creating a billing project against an billing account with no access" in {
+    val samDAO = mock[SamDAO]
+    val createRequest = CreateRawlsV2BillingProjectFullRequest(
+      RawlsBillingProjectName("fake_project_name"),
+      Some(RawlsBillingAccountName("fake_billing_account_name")),
+      None,
+      None
+    )
+    val gcsDAO = mock[GoogleServicesDAO]
+    when(gcsDAO.testBillingAccountAccess(
+      ArgumentMatchers.eq(createRequest.billingAccount.get),
+      ArgumentMatchers.eq(userInfo))
+    ).thenReturn(Future.successful(false))
+    val gbp = new GoogleBillingProjectCreator(samDAO, gcsDAO)
+
+    val ex = intercept[GoogleBillingAccountAccessException] {
+      Await.result(gbp.validateBillingProjectCreationRequest(createRequest, userInfo), Duration.Inf)
+    }
+
+    assertResult(Some(StatusCodes.BadRequest)) {
+      ex.errorReport.statusCode
+    }
+  }
+
+  it should "fail when provided with a service perimeter name but no access" in {
+    val servicePerimeterName = ServicePerimeterName("fake_sp_name")
+    val samDAO = mock[SamDAO]
+
+    when(
+      samDAO.userHasAction(
+        ArgumentMatchers.eq(SamResourceTypeNames.servicePerimeter),
+        ArgumentMatchers.eq(servicePerimeterName.value),
+        ArgumentMatchers.eq(SamServicePerimeterActions.addProject),
+        ArgumentMatchers.eq(userInfo))).thenReturn(Future.successful(false))
+
+    val createRequest = CreateRawlsV2BillingProjectFullRequest(
+      RawlsBillingProjectName("fake_billing_project"),
+      Some(RawlsBillingAccountName("fake_billing_account_name")),
+      Some(servicePerimeterName),
+      None
+    )
+    val bpo = new GoogleBillingProjectCreator(
+      samDAO,
+      mock[GoogleServicesDAO],
+    )
+
+    val ex = intercept[ServicePerimeterAccessException] {
+      Await.result(bpo.validateBillingProjectCreationRequest(createRequest, userInfo), Duration.Inf)
+    }
+
+    assertResult(Some(StatusCodes.Forbidden)) {
+      ex.errorReport.statusCode
+    }
+  }
+
+  behavior of "postCreationSteps"
+
+  it should "sync the policy to google" in {
+    val samDAO = mock[SamDAO]
+    val createRequest = CreateRawlsV2BillingProjectFullRequest(
+      RawlsBillingProjectName("fake_project_name"),
+      Some(RawlsBillingAccountName("fake_billing_account_name")),
+      None,
+      None
+    )
+    when(
+      samDAO.syncPolicyToGoogle(
+        ArgumentMatchers.eq(SamResourceTypeNames.billingProject),
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner))
+    ).thenReturn(Future.successful(Map(WorkbenchEmail(userInfo.userEmail.value) -> Seq())))
+    val gbp = new GoogleBillingProjectCreator(samDAO, mock[GoogleServicesDAO])
+
+    Await.result(gbp.postCreationSteps(createRequest, userInfo), Duration.Inf)
```

Comment:
Is the assertion missing in this test or is `postCreationSteps` not throwing a sufficient assertion? Could add a `verify` that `syncPolicyToGoogle` gets called

Reply:
Good catch, will add that verification.